src/components/data-exploration/differential-expression-tool/AdvancedFilteringModal.jsx

```diff
@@ -0,0 +1,162 @@
+import React, { useState } from 'react';
+import {
+  Modal, Form, Button, Space, Select, InputNumber, Dropdown, Menu,
+} from 'antd';
+import PropTypes from 'prop-types';
+import { CloseOutlined, PlusOutlined } from '@ant-design/icons';
+
+const AdvancedFilteringModal = (props) => {
+  const { onCancel } = props;
+  const [form] = Form.useForm();
+  const [valuesDomain, setValuesDomain] = useState([]);
+
+  const criteriaOptions = [
+    { value: 'logfc', label: 'logFC' },
+    { value: 'pValue', label: 'adj p-value' },
+    { value: 'pct1', label: 'Pct1' },
+    { value: 'pct2', label: 'Pct2' },
+    { value: 'auc', label: 'AUC' },
+  ];
+
+  const conditionOptions = [{
+    value: 'gt',
+    label: 'Greater than',
+  }, {
+    value: 'lt',
+    label: 'Less than',
+  },
+  ];
+
+  const valueRestrictions = {
+    pct1: [0, 100],
+    pct2: [0, 100],
+    logfc: [-50, 50],
+    auc: [0, 1],
+    pValue: [0, 1],
+  };
+
+  const presetFilters = {
+    'Up-regulated': {
+      criteria: 'logfc',
+      condition: 'gt',
+      value: 0,
+    },
+    'Down-regulated': {
+      criteria: 'logfc',
+      condition: 'lt',
+      value: 0,
+    },
+    Significant: {
+      criteria: 'pValue',
+      condition: 'lt',
+      value: 0.05,
+    },
+  };
+  const renderPresetFilters = (add) => (
+    <Menu
+      onClick={(e) => {
+        add(presetFilters[e.key]);
+        changeCriteria(presetFilters[e.key].criteria);
+      }}
+    >
+      {Object.keys(presetFilters).map((filter) => (
+        <Menu.Item key={filter}>
+          {filter}
+        </Menu.Item>
+      ))}
+    </Menu>
+  );
+
+  const changeCriteria = (criteria, index = false) => {
```

Comment:
It better to use `null` for elements that aren't set, javascript allows us to do this but it is usually cleaner to avoid having many different possible types in a single variable.